src/recommendation/openShiftToolkit.ts

```diff
@@ -39,7 +39,7 @@ export function initializeRecommendation(context: vscode.ExtensionContext, handl
     })
   );
 
-  const isdevfileYAMLOpened = vscode.workspace.workspaceFolders.findIndex((workspace) => isDevfileYAML(workspace.uri)) !== -1;
+  const isdevfileYAMLOpened = vscode.workspace.workspaceFolders?.findIndex((workspace) => isDevfileYAML(workspace.uri)) === 0;
```

Comment:
use `> -1` instead of `=== 0`.  There is no guarantee that devfile will be on index 0.